tensorflow/core/kernels/softmax_op_gpu.cu.cc

```diff
@@ -52,12 +52,24 @@ __device__ __host__ EIGEN_STRONG_INLINE float strict_cast<float, Eigen::half>(
   return functor::HalfToFloat<Eigen::half>()(t);
 }
 
+template <>
+__device__ __host__ EIGEN_STRONG_INLINE float strict_cast<float, Eigen::bfloat16>(
+    Eigen::bfloat16 t) {
+  return functor::HalfToFloat<Eigen::bfloat16>()(t);
```

Comment:
This is now just a `static_cast<float>(...)`, so you should prefer using that.

The original implementation of `HalfToFloat` was specifically only written for `Eigen::half` and relied on internal implementation details, so this wouldn't have worked for `bfloat16`.  In fact, we should probably remove both `HalfToFloat` and `FloatToHalf` altogether and use `static_cast` in all cases.

Reply:
Thanks @cantonios for reviewing! That make sense - will fix.

Reply:
I ended up also removing `strict_cast` entirely since static_cast from float to float is a no-op anyway - do you see any problems with this?

Reply:
I don't off-hand.  I'm not sure what the original intention of that was - if it was really meant to enforce some type of strict casting behavior, or if it was just a middleman for allowing specializations for half.